tests/tracer/test_tracer.py

```diff
@@ -492,6 +498,52 @@ def test_configure_dogstatsd_url_socket(self):
         assert tracer._writer.dogstatsd.port is None
         assert tracer._writer.dogstatsd.socket_path == "/foo.sock"
 
+    def test_tracer_set_user(self):
+        span = self.trace("fake_span")
+        set_user(
+            self.tracer,
+            user_id="usr.id",
+            email="usr.email",
+            name="usr.name",
+            session_id="usr.session_id",
+            role="usr.role",
+            scope="usr.scope",
+        )
+        assert span.get_tag(user.ID)
+        assert span.get_tag(user.EMAIL)
+        assert span.get_tag(user.SESSION_ID)
+        assert span.get_tag(user.NAME)
+        assert span.get_tag(user.ROLE)
+        assert span.get_tag(user.SCOPE)
+
+    def test_tracer_set_user_mandatory(self):
+        span = self.trace("fake_span")
+        set_user(
+            self.tracer,
+            user_id="usr.id",
+        )
+        span_keys = list(span.get_tags().keys())
+        span_keys.sort()
+        assert span_keys == ["runtime-id", "usr.id"]
+        assert span.get_tag(user.ID)
+        assert span.get_tag(user.EMAIL) is None
+        assert span.get_tag(user.SESSION_ID) is None
+        assert span.get_tag(user.NAME) is None
+        assert span.get_tag(user.ROLE) is None
+        assert span.get_tag(user.SCOPE) is None
+
+    def test_tracer_set_user_warning_no_span(self):
+        class MockTracerNoSpan:
+            def current_root_span(self):
+                return None
+
+        with self._caplog.at_level(logging.WARNING):
+            set_user(
+                MockTracerNoSpan(),
```

Comment:
we could probably just use `self.tracer` here, as long as we don't start a span there shouldn't be a current root span. it would be better to test with a "real" tracer so we don't need to worry about updating this, or if something changes about `current_root_span` this test will (hopefully) fail.